src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -8114,6 +8104,35 @@ namespace VariableSpeedCoils {
         return Speeds;
     }
 
+    Real64 GetVSCoilRatedSourceTemp(EnergyPlusData &state,
+                                    int const CoilIndex, // index to cooling coil
+                                    bool &ErrorsFound    // set to true if problem
+    )
+    {
+        Real64 RatedSourceTemp = 0.0;
+        switch (state.dataVariableSpeedCoils->VarSpeedCoil(CoilIndex).VSCoilType) {
+        case DataHVACGlobals::Coil_CoolingWaterToAirHPVSEquationFit: {
+            RatedSourceTemp = RatedInletWaterTemp;
+        }
+        case DataHVACGlobals::Coil_HeatingWaterToAirHPVSEquationFit: {
+            RatedSourceTemp = RatedInletWaterTempHeat;
+        }
+        case DataHVACGlobals::CoilDX_HeatPumpWaterHeaterVariableSpeed: {
+            RatedSourceTemp = state.dataVariableSpeedCoils->VarSpeedCoil(CoilIndex).WHRatedInletWaterTemp;
+        }
+        case DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed: {
+            RatedSourceTemp = RatedAmbAirTemp;
+        }
+        case DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed: {
+            RatedSourceTemp = RatedAmbAirTempHeat;
+        }
+        default: {
+            assert(false);
+        }
+        }
+        return RatedSourceTemp;
+    }
```

Comment:
You didn't need to change the returns. And I just noticed you didn't have break at the each of each case. So 1 more correction to fix. @amirroth if there is a return for each case, do you even need the break statement?

Reply:
Oops. Thanks for pointing it out, I just added them.